tests/foreman/cli/test_computeresource_vmware.py

```diff
@@ -11,158 +11,128 @@
             'provider': FOREMAN_PROVIDERS['vmware'],
-            'server': vmware.vmware_server,
-            'user': vmware.vmware_username,
-            'password': vmware.vmware_password,
-            'datacenter': vmware.vmware_datacenter,
+            'server': settings.vmware.vcenter,
+            'user': settings.vmware.username,
+            'password': settings.vmware.password,
+            'datacenter': settings.vmware.datacenter,
         }
     )
     assert vmware_cr['name'] == cr_name
-
-
-@pytest.mark.tier1
-def test_positive_create_with_loc(vmware):
-    """Create VMware Compute Resource with locations
-
-    :id: 214a0f54-6fc2-4e7b-91ab-a45760ffb2f2
-
-    :customerscenario: true
-
-    :expectedresults: VMware Compute resource is created
-
-    :BZ: 1387917
+    assert vmware_cr['locations'][0] == module_location.name
+    assert vmware_cr['organizations'] == module_org.name
+    assert vmware_cr['server'] == settings.vmware.vcenter
+    assert vmware_cr['datacenter'] == settings.vmware.datacenter
+
+
+@pytest.mark.e2e
+@pytest.mark.on_premises_provisioning
+@pytest.mark.parametrize('setting_update', ['destroy_vm_on_host_delete=True'], indirect=True)
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+@pytest.mark.parametrize('provision_method', ['build', 'bootdisk'])
+@pytest.mark.rhel_ver_match('[^6]')
+@pytest.mark.tier3
+def test_positive_provision_end_to_end(
+    request,
+    setting_update,
+    module_provisioning_sat,
+    module_sca_manifest_org,
+    module_location,
+    pxe_loader,
+    module_vmware_cr,
+    module_vmware_hostgroup,
+    provision_method,
+):
+    """Provision a host on vmware compute resource with
+    the help of hostgroup.
+
+    :id: ff9963fc-a2a7-4392-aa9a-190d5d1c8357
+
+    :steps:
+
+        1. Configure provisioning setup.
+        2. Create VMware CR
+        3. Configure host group setup.
+        4. Provision a host on VMware
+        5. Verify created host on VMware with wrapanapi
+
+    :expectedresults: Host is provisioned succesfully with hostgroup
 
     :CaseAutomation: Automated
-
-    :CaseImportance: Critical
     """
-    cr_name = gen_string('alpha')
-    vmware_cr = make_compute_resource(
+    sat = module_provisioning_sat.sat
+    hostname = gen_string('alpha').lower()
+    host = sat.cli.Host.create(
         {
-            'name': cr_name,
-            'location-ids': vmware.loc.id,
-            'provider': FOREMAN_PROVIDERS['vmware'],
-            'server': vmware.vmware_server,
-            'user': vmware.vmware_username,
-            'password': vmware.vmware_password,
-            'datacenter': vmware.vmware_datacenter,
+            'name': hostname,
+            'organization': module_sca_manifest_org.name,
+            'location': module_location.name,
+            'hostgroup': module_vmware_hostgroup.name,
+            'compute-resource-id': module_vmware_cr.id,
+            'ip': None,
+            'mac': None,
```

Comment:
Quick question: Are we keeping both of these none for later alignment post the provisioning process? Can we use something like gen_ipaddr()/gen_mac() instead of none ... thoughts @shubhamsg199 ?

Reply:
We are intentionally keeping it none just to make sure no random values are assigned to them